File: Llama 3.1 70B/C++/2-4.c
// Binary Search Program

#include <iostream>
using namespace std;

int binarySearch(int arr[], int n, int target) {
    int low = 0;
    int high = n - 1;

    while (low <= high) {
        int mid = (low + high) / 2;
        if (arr[mid] == target) {
            return mid;
        } else if (arr[mid] < target) {
            low = mid + 1;
        } else {
            high = mid - 1;
        }
    }

    return -1;
}

int main() {
    int n;
    cout << "Binary Search Program" << endl;
    cout << "Enter the size of the array: ";
    cin >> n;

    int arr[n];
    cout << "Enter the elements of the array in ascending order: ";
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    int target;
    cout << "Enter the target value: ";
    cin >> target;

    int index = binarySearch(arr, n, target);
    if (index != -1) {
        cout << "Target value found at index " << index << endl;
    } else {
        cout << "Target value not found" << endl;
    }

    return 0;
}

File: Llama 3.1 70B/Python/2-4.py
# Binary Search Program

def binary_search(arr, target):
    """Returns the index of the target value in a sorted list."""
    low, high = 0, len(arr) - 1

    while low <= high:
        mid = (low + high) // 2
        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            low = mid + 1
        else:
            high = mid - 1

    return -1  # Target value not found

def main():
    print("Binary Search Program")
    arr = [int(x) for x in input("Enter a sorted list of numbers (space-separated): ").split()]
    target = int(input("Enter the target value: "))

    index = binary_search(arr, target)
    if index != -1:
        print("Target value found at index", index)
    else:
        print("Target value not found")

if __name__ == "__main__":
    main()

